extensions/guacamole-auth-radius/src/main/java/org/apache/guacamole/auth/radius/RadiusConnectionService.java

```diff
@@ -0,0 +1,394 @@
+            logger.debug("Failed to init RADIUS client.", e);
+            return;
+        }
+        catch (UnknownHostException e) {
+            logger.error("Unable to resolve host: {}", e.getMessage());
+            logger.debug("Failed to resolve host.", e);
+            return;
+        }
+        catch (IOException e) {
+            logger.error("Unable to communicate with host: {}", e.getMessage());
+            logger.debug("Failed to communicate with host.", e);
+            return;
+        }
+
+    }
+
+    /**
+     * Creates a new instance of RadiusAuthentictor, configured with
+     * parameters specified within guacamole.properties.
+     *
+     * @return
+     *     A new RadiusAuthenticator instance which has been configured
+     *     with parameters from guacamole.properties, or null if
+     *     configuration fails.
+     *
+     */
+    private RadiusAuthenticator setupRadiusAuthenticator() {
+
+        // If we don't have a radiusClient object, yet, don't go any further.
+        if (radiusClient == null) {
+            logger.error("RADIUS client hasn't been set up, yet.");
+            logger.debug("We can't run this method until the RADIUS client has been set up.");
+            return null;
+        }
+
+        String radAuthName;
+        String caFile;
+        String caPassword;
+        String caType;
+        String keyFile;
+        String keyPassword;
+        String keyType;
+        Boolean trustAll;
+        String innerProtocol;
+        LocalEnvironment guacEnv;
+        String basePath;
+
+        // Pull configuration parameters from guacamole.properties
+        try {
+            guacEnv = new LocalEnvironment();
+            basePath = guacEnv.getGuacamoleHome().getAbsolutePath() + '/';
+            radAuthName = confService.getRadiusAuthProtocol();
+            caFile = confService.getRadiusCAFile();
+            caPassword = confService.getRadiusCAPassword();
+            caType = confService.getRadiusCAType();
+            keyFile = confService.getRadiusKeyFile();
+            keyPassword = confService.getRadiusKeyPassword();
+            keyType = confService.getRadiusKeyType();
+            trustAll = confService.getRadiusTrustAll();
+            innerProtocol = confService.getRadiusEAPTTLSInnerProtocol();
+            
+        }
+        catch (GuacamoleException e) {
+            logger.error("Error retrieving configuration.");
+            logger.debug("Error getting config parameters from file.");
+            return null;
+        }
+
+        RadiusAuthenticator radAuth = radiusClient.getAuthProtocol(radAuthName);
+        if (radAuth == null)
+            return null;
+
+        // If we're using any of the TLS protocols, we need to configure them
+        if (radAuth instanceof PEAPAuthenticator || 
+            radAuth instanceof EAPTLSAuthenticator || 
+            radAuth instanceof EAPTTLSAuthenticator) {
+
+            if (caFile != null && !caFile.isEmpty())
```

Comment:
I removed the isEmpty() checks from this, but I'm not sure removing the null checks from this section would really work.  It is perfectly okay to leave out some of these values in the configuration, and, since we're doing the actual RADIUS library calls here, we want to make sure we don't try to send a null value into one of those methods.  Some of them should never be null - for example keyType has a value if not specified - so I can remove those - but others, like the caPassword and keyPassword, could be null and we probably shouldn't try to send a null value, right?